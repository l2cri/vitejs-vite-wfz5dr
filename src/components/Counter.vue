<script setup>
import { ref, computed, watch } from 'vue';
import { useCounter } from '../composables/counter.js';

const props = defineProps({
  count: { type: Number, default: 0 },
});

const { counter, increase, decrease } = useCounter(props.count);
const biggest = ref(false);

const doubleCount = computed(() => counter.value * 2);
watch(
  () => counter.value,
  (value) => {
    if (value > 10) {
      biggest.value = true;
    } else {
      biggest.value = false;
    }
  },
  { immediate: true }
);
</script>

<template>
  <div>Count : {{ counter }}</div>
  <div>DoubleCount : {{ doubleCount }}</div>

  <button @click="increase">increment</button>
  <button @click="decrease">decrement</button>
  <h2 v-if="biggest">More than 10</h2>
</template>
