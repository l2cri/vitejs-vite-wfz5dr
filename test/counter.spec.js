import { mount } from '@vue/test-utils';
import Counter from '../src/components/Counter.vue';

test('mount component', async () => {
  expect(Counter).toBeTruthy();

  const wrapper = mount(Counter, {
    props: { count: 9 },
  });

  expect(wrapper.text()).toContain('Count : 9');
  expect(wrapper.html()).toMatchSnapshot();

  await wrapper.get('button').trigger('click');

  expect(wrapper.text()).toContain('Count : 10');

  await wrapper.get('button').trigger('click');

  expect(wrapper.text()).toContain('Count : 11');
  expect(wrapper.text()).toContain('More than 10');
});
