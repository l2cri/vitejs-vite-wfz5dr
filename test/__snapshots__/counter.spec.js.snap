// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div>Count : 9</div>
<div>DoubleCount : 18</div>
<button>increment</button>
<button>decrement</button>
<!--v-if-->"
`;
